File: src/main/java/org/springframework/samples/petclinic/rest/mapper/VetMapper.java
package org.springframework.samples.petclinic.rest.mapper;

import org.mapstruct.Mapper;
import org.springframework.samples.petclinic.vet.Vet;
import org.springframework.samples.petclinic.rest.dto.VetDto;

@Mapper(componentModel = "spring", uses = { SpecialtyMapper.class })
public interface VetMapper {

	VetDto toDto(Vet vet);

}

File: src/main/kotlin/org/springframework/samples/petclinic/rest/mapper/VetMapper.kt
package org.springframework.samples.petclinic.rest.mapper

import org.springframework.samples.petclinic.rest.dto.VetDto
import org.springframework.samples.petclinic.vet.Vet
import org.springframework.stereotype.Component

@Component
class VetMapper(
    private val specialtyMapper: SpecialtyMapper
) {

    fun toDto(vet: Vet): VetDto {
        return VetDto(
            id = vet.id,
            firstName = vet.firstName,
            lastName = vet.lastName,
            specialties = vet.getSpecialties().map { specialtyMapper.toDto(it) }
        )
    }

}